vlib/v/ast/README.md

```diff
@@ -0,0 +1,2837 @@
+# Vlang abstract syntax tree(AST) introduction
+
+## Overview
+
+The Vlang abstract syntax tree is implemented by sum type. the following is declaration:
```

Comment:
`by using sum types`.

Also put a link to vlib/v/ast/ast.v , do not duplicate the declarations here too - it would be harder to keep them in sync, there should be one source of truth, and that is ast.v, which the compiler also uses, unlike the README.md which may/will get stale.